### ISSUE TITLE
Service won't start without -disablewallet flag

### ISSUE THREAD

**Issue by djadk84** [dashpay/dash#629]
Im currently using freebsd 10.1 RELEASE I install dash coin via pkgsrc when I try to start service
with _darkcoind -daemon_

************************
darkcoind -daemon
EXCEPTION: N5boost10filesystem16filesystem_errorE       
boost::filesystem::copy_file: No such file or directory: "/home/dash/.dash/wallet.dat", "/home/dash/.dash/backups/wallet.dat.2015-09-28-21.27"       
dash in AppInit()

2015-09-28 21:34:45 Dash version v0.12.0.0-gc489574 (Sun, 14 Jun 2015 17:05:51 -0700)
2015-09-28 21:34:45 Using OpenSSL version OpenSSL 1.0.2d 9 Jul 2015
2015-09-28 21:34:45 Using BerkeleyDB version Berkeley DB 4.8.30: (April  9, 2010)
2015-09-28 21:34:45 Default data directory /home/darkcoind/.dash
2015-09-28 21:34:45 Using data directory /home/darkcoind/.dash
2015-09-28 21:34:45 Using config file /home/darkcoind/.dash/dash.conf
2015-09-28 21:34:45 Using at most 125 connections (116388 file descriptors available)
2015-09-28 21:34:45 Using 8 threads for script verification
2015-09-28 21:34:45 

************************
EXCEPTION: N5boost10filesystem16filesystem_errorE       
boost::filesystem::copy_file: No such file or directory: "/home/darkcoind/.dash/wallet.dat", "/home/darkcoind/.dash/backups/wallet.dat.2015-09-28-21.34"       
dash in AppInit()       
cat debug.log
2015-09-28 21:34:45 PrepareShutdown: In progress...
2015-09-28 21:34:45 StopNode()
2015-09-28 21:34:45 Verifying mncache.dat format...
2015-09-28 21:34:45 Loaded info from mncache.dat  0ms
2015-09-28 21:34:45   Masternodes: 0, peers who asked us for Masternode list: 0, peers we asked for Masternode list: 0, entries in Masternode list we asked for: 0, nDsqCount: 0
2015-09-28 21:34:45 Writting info to mncache.dat...
2015-09-28 21:34:45 Written info to mncache.dat  0ms
2015-09-28 21:34:45   Masternodes: 0, peers who asked us for Masternode list: 0, peers we asked for Masternode list: 0, entries in Masternode list we asked for: 0, nDsqCount: 0
2015-09-28 21:34:45 Masternode dump finished  1ms
2015-09-28 21:34:45 Verifying budget.dat format...
2015-09-28 21:34:45 Loaded info from budget.dat  0ms
2015-09-28 21:34:45   not implemented
2015-09-28 21:34:45 Writting info to budget.dat...
2015-09-28 21:34:45 Written info to budget.dat  0ms
2015-09-28 21:34:45 Masternode dump finished  0ms
2015-09-28 21:34:45 Shutdown: done

**Comment by UdjinM6** [comment]
This version "Dash version v0.12.0.0-gc489574 (Sun, 14 Jun 2015 17:05:51 -0700)" is completely outdated and should not be used. Build new one (v0.12.0.55) from source (master branch) pls.

**Comment by djadk84** [comment]
$ darkcoind -daemon
Tue Sep 29 07:54:48 AST 2015
darkcoind is not running
Dash server starting
$ 

************************
EXCEPTION: N5boost10filesystem16filesystem_errorE       
boost::filesystem::copy_file: No such file or directory: "/home/darkcoind/.dash/testnet3/wallet.dat", "/home/darkcoind/.dash/testnet3/backups/wallet.dat.2015-09-29-11-54"       
dash in AppInit()       

$ cat testnet3/debug.log
2015-09-29 11:54:48 AppInit2 : parameter interaction: -bind or -whitebind set -> setting -listen=1
2015-09-29 11:54:48 
2015-09-29 11:54:48 Dash version v0.12.0.55-gc30a0aa (Mon, 28 Sep 2015 17:49:30 +0200)
2015-09-29 11:54:48 Using OpenSSL version OpenSSL 1.0.2d 9 Jul 2015
2015-09-29 11:54:48 Using BerkeleyDB version Berkeley DB 4.8.30: (April  9, 2010)
2015-09-29 11:54:48 Default data directory /home/darkcoind/.dash
2015-09-29 11:54:48 Using data directory /home/darkcoind/.dash/testnet3
2015-09-29 11:54:48 Using config file /home/darkcoind/.dash/dash.conf
2015-09-29 11:54:48 Using at most 512 connections (116388 file descriptors available)
2015-09-29 11:54:48 Using 8 threads for script verification
2015-09-29 11:54:48 Binding RPC on address 192.168.1.15 port 18335 (IPv4+IPv6 bind any: 0)
2015-09-29 11:54:48 

************************
EXCEPTION: N5boost10filesystem16filesystem_errorE       
boost::filesystem::copy_file: No such file or directory: "/home/darkcoind/.dash/testnet3/wallet.dat", "/home/darkcoind/.dash/testnet3/backups/wallet.dat.2015-09-29-11-54"       
dash in AppInit()       

2015-09-29 11:54:48 PrepareShutdown: In progress...
2015-09-29 11:54:48 StopNode()
2015-09-29 11:54:48 Verifying mncache.dat format...
2015-09-29 11:54:48 ERROR: Read : Failed to open file /home/darkcoind/.dash/testnet3/mncache.dat
2015-09-29 11:54:48 Missing masternode cache file - mncache.dat, will try to recreate
2015-09-29 11:54:48 Writting info to mncache.dat...
2015-09-29 11:54:48 Written info to mncache.dat  0ms
2015-09-29 11:54:48   Masternodes: 0, peers who asked us for Masternode list: 0, peers we asked for Masternode list: 0, entries in Masternode list we asked for: 0, nDsqCount: 0
2015-09-29 11:54:48 Masternode dump finished  0ms
2015-09-29 11:54:48 Verifying budget.dat format...
2015-09-29 11:54:48 ERROR: Read : Failed to open file /home/darkcoind/.dash/testnet3/budget.dat
2015-09-29 11:54:48 Missing budgets file - budget.dat, will try to recreate
2015-09-29 11:54:48 Writting info to budget.dat...
2015-09-29 11:54:48 Written info to budget.dat  0ms
2015-09-29 11:54:48 Budget dump finished  1ms
2015-09-29 11:54:48 Verifying mnpayments.dat format...
2015-09-29 11:54:48 ERROR: Read : Failed to open file /home/darkcoind/.dash/testnet3/mnpayments.dat
2015-09-29 11:54:48 Missing budgets file - mnpayments.dat, will try to recreate
2015-09-29 11:54:48 Writting info to mnpayments.dat...
2015-09-29 11:54:48 Written info to mnpayments.dat  0ms
2015-09-29 11:54:48 Budget dump finished  0ms
2015-09-29 11:54:48 Shutdown: done

**Comment by UdjinM6** [comment]
hmm... ok. 
1) Why are you running `darkcoind`? There should be `dashd` daemon now.
2) Is there `/home/darkcoind/.dash` folder at all?
3) Also check user you running it under and his permissions.
4) Is there any specific scripts/configs you made earlier to run it?

**Comment by djadk84** [comment]
1) Why are you running darkcoind? There should be dashd daemon now.
Yes I know, I used to run the v0.9.12.32 under pkgsrc now I'm trying to update it to the v0.12.0.55 and I haven't the binary name from darkcoind to dashd 
2) Is there /home/darkcoind/.dash folder at all?
Yes there is.
$ pwd
home/darkcoind/.dash/testnet3
$ ls
backups		budget.dat	dashd.pid	masternode.conf	mnpayments.dat
blocks		chainstate	debug.log	mncache.dat	peers.dat
3) Also check user you running it under and his permissions.
$ ls -l
total 25
drwx------  2 darkcoind  darkcoind      2 Sep 29 07:54 backups
drwx------  3 darkcoind  darkcoind      5 Sep 29 08:02 blocks
-rw-------  1 darkcoind  darkcoind     61 Sep 29 07:54 budget.dat
drwx------  2 darkcoind  darkcoind      7 Sep 29 08:02 chainstate
-rw-------  1 darkcoind  darkcoind      6 Sep 29 08:02 dashd.pid
-rw-------  1 darkcoind  darkcoind  12739 Sep 29 08:55 debug.log
-rw-r--r--  1 darkcoind  darkcoind    264 Sep 29 07:54 masternode.conf
-rw-------  1 darkcoind  darkcoind     66 Sep 29 07:54 mncache.dat
-rw-------  1 darkcoind  darkcoind     57 Sep 29 07:54 mnpayments.dat
-rw-------  1 darkcoind  darkcoind   4508 Sep 29 08:47 peers.dat
$ cd ..
$ ls -l
total 239
-rw-------  1 darkcoind  darkcoind     552 Sep 29 00:49 dash.conf
drwxr-xr-x  5 darkcoind  darkcoind      13 Sep 29 08:17 testnet3
$ ls -ld .dash/
drwxr-xr-x  3 darkcoind  darkcoind  9 Sep 29 07:54 .dash/
$ whoami
darkcoind
4) Is there any specific scripts/configs you made earlier to run it?
$ vi .dash/dash.conf 

rpcuser=dashuser
rpcpassword=dashpassword
testnet=1
server=1
rpctimeout=30
txindex=1
maxconnections=512
rpcport=PORT
port=PORT
rpcallowip=IP_
bind=IP_
rpcbind=IP_

I'm running a freebsd 10.1 with pkgsrc with clang compiler.

**Comment by UdjinM6** [comment]
that's weird..
apply this patch, recompile and try again pls
```
diff --git a/src/init.cpp b/src/init.cpp
index db615b4..deac30e 100644
--- a/src/init.cpp
+++ b/src/init.cpp
@@ -914,11 +914,13 @@ bool AppInit2(boost::thread_group& threadGroup)
                 boost::filesystem::path backupFile = backupPathStr + dateTimeStr;
                 sourceFile.make_preferred();
                 backupFile.make_preferred();
-                try {
-                    boost::filesystem::copy_file(sourceFile, backupFile);
-                    LogPrintf("Creating backup of %s -> %s\n", sourceFile, backupFile);
-                } catch(boost::filesystem::filesystem_error &error) {
-                    LogPrintf("Failed to create backup %s\n", error.what());
+                if(boost::filesystem::exists(sourceFile)) {
+                    try {
+                        boost::filesystem::copy_file(sourceFile, backupFile);
+                        LogPrintf("Creating backup of %s -> %s\n", sourceFile, backupFile);
+                    } catch(boost::filesystem::filesystem_error &error) {
+                        LogPrintf("Failed to create backup %s\n", error.what());
+                    }
                 }
                 // Keep only the last 10 backups, including the new one of course
                 typedef std::multimap<std::time_t, boost::filesystem::path> folder_set_t;
```

**Comment by djadk84** [comment]
Works, no more error. Will you be applying those changes to master so I dont have to maintain this patch locally? Please let me know.
Thanks, I really appreciate your time in solving this issue.

**Comment by UdjinM6** [comment]
Cool! Glad it solved an issue, made a pull request to 0.12.0.x branch for other devs to review it (and take a note if there smth like this in the future).
Thanks for reporting!

**Comment by UdjinM6** [comment]
merged in 0.12.0.x, closing
